fe/fe-core/src/main/java/com/starrocks/sql/analyzer/PrivilegeCheckerV2.java

```diff
@@ -180,6 +207,81 @@ static void checkStmtNodePrivilege(ConnectContext context) {
         }
     }
 
+    public static Set<TableName> getAllTableNamesForAnalyzeJobStmt(long dbId, long tableId) {
+        Set<TableName> tableNames = Sets.newHashSet();
+        if (StatsConstants.DEFAULT_ALL_ID != tableId && StatsConstants.DEFAULT_ALL_ID != dbId) {
+            Database db = GlobalStateMgr.getCurrentState().getDb(dbId);
+            if (db != null && !db.isInfoSchemaDb()) {
+                Table table = db.getTable(tableId);
+                if (table != null) {
+                    tableNames.add(new TableName(InternalCatalog.DEFAULT_INTERNAL_CATALOG_NAME,
+                            db.getFullName(), table.getName()));
+                }
+            }
+        } else if (StatsConstants.DEFAULT_ALL_ID == tableId && StatsConstants.DEFAULT_ALL_ID != dbId) {
+            addTableNamesInDb(tableNames, dbId);
+        } else if (dbId == StatsConstants.DEFAULT_ALL_ID) {
+            List<Long> dbIds = GlobalStateMgr.getCurrentState().getDbIds();
+            for (Long id : dbIds) {
+                addTableNamesInDb(tableNames, id);
+            }
+        }
+
+        return tableNames;
+    }
+
+    private static void addTableNamesInDb(Set<TableName> tableNames, Long id) {
```

Comment:
How about name as `getTableNamesInDb`? `addTableNamesInDb` seems a ddl operation.

Reply:
good idea, actually this is a function generated by Intellij and i didn't look it carefully...